front/pages/poke/[wId]/upgrade_enterprise/index.tsx

```diff
@@ -0,0 +1,303 @@
+import { PokeButton } from "@app/components/poke/shadcn/ui/button";
+import { PokeCheckbox } from "@app/components/poke/shadcn/ui/checkbox";
+import {
+  PokeForm,
+  PokeFormControl,
+  PokeFormField,
+  PokeFormItem,
+  PokeFormLabel,
+  PokeFormMessage,
+} from "@app/components/poke/shadcn/ui/form";
+import { PokeInput } from "@app/components/poke/shadcn/ui/input";
+import { withSuperUserAuthRequirements } from "@app/lib/iam/session";
+
+export const getServerSideProps = withSuperUserAuthRequirements<{
+  owner: WorkspaceType;
+}>(async (context, auth) => {
+  const owner = auth.workspace();
+  if (!owner) {
+    return {
+      notFound: true,
+    };
+  }
+
+  return {
+    props: {
+      owner,
+    },
+  };
+});
+
+export default function NewEnterpriseSubscriptionForm({
+  owner,
+}: {
+  owner: WorkspaceType;
+}) {
+  const router = useRouter();
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [error, setError] = React.useState<string | null>(null);
+
+  const form = useForm<NewEnterpriseSubscriptionType>({
+    resolver: ioTsResolver(NewEnterpriseSubscriptionSchema),
+    defaultValues: {
+      stripeSubscriptionId: "",
+      code: "",
+      name: "",
+      isSlackbotAllowed: true,
+      isSlackAllowed: true,
+      isNotionAllowed: true,
+      isGoogleDriveAllowed: true,
+      isGithubAllowed: true,
+      isIntercomAllowed: true,
+      isConfluenceAllowed: true,
+      isWebCrawlerAllowed: true,
+      maxMessages: -1,
+      dataSourcesCount: -1,
+      dataSourcesDocumentsCount: -1,
+      dataSourcesDocumentsSizeMb: 2,
+      maxUsers: 1000,
+    },
+  });
+
+  const onSubmit = (values: NewEnterpriseSubscriptionType) => {
+    const cleanedValues = Object.fromEntries(
+      removeNulls(
+        Object.entries(values).map(([key, value]) => {
+          if (typeof value !== "string") {
+            return [key, value];
+          }
+          const cleanedValue = value.trim();
+          if (!cleanedValue) {
+            return null;
+          }
+          return [key, cleanedValue];
+        })
+      )
+    );
+
+    void submit();
+    async function submit() {
+      setIsSubmitting(true);
+      try {
+        const r = await fetch(
+          `/api/poke/workspaces/${owner.sId}/upgrade_enterprise`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify(cleanedValues),
+          }
+        );
+
+        if (!r.ok) {
+          throw new Error(
+            `Something went wrong: ${r.status} ${await r.text()}`
+          );
+        }
+
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+        void router.push(`/poke/${owner.sId}`);
+      } catch (e) {
+        setIsSubmitting(false);
+        if (e instanceof Error) {
+          setError(e.message);
+        }
+      }
+    }
+  };
+
+  console.log("Form errors", form.formState.errors);
```

Comment:
✂️

Reply:
I _think_ it's useful because I lost some time not understanding why my form was not submitting and there was a hidden error in here. I should have added a comment to make it clear tho, will do that 😄

Reply:
Can we log only if we have some errors then?